notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "Su5oVLIL-GK5"
+   },
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "import sklearn\n",
+    "\n",
+    "%matplotlib inline\n",
+    "warnings.simplefilter(action='ignore', category=FutureWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 237
+    },
+    "id": "1jUWQVXZ-HOs",
+    "outputId": "28317993-f1b2-44c1-d468-1fd00fb58f0f",
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "train_data = pd.read_csv('C:\\\\Users\\\\Owner\\\\Desktop\\\\Proga\\\\REPOS\\\\mentorship_EPAM\\\\data\\\\train.csv', index_col = \"id\")\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.tail()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.info()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"] = train_data[\"family\"].str.lower()     # for easier reading \n",
+    "train_data.describe(percentiles=[.25, 0.375, .5, .75, .875]).drop([\"count\"])       \n",
+    "\n",
+    "# I dropped \"count\" row to get rid of exponentioal number presentation\n",
+    "# And added .875 percentile to understand approximately how many objects have no onpromotion"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "6yDCEDYYGveo",
+    "outputId": "839affcc-2928-4892-c798-459ff56b59f0"
+   },
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>The only categorical feature (in this table) \"family\" has 33 possible unique values (not so a lot) <br>\n",
+    "That means we can easily use one-hot-encoding during model training</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "family_values = train_data[\"family\"].unique()\n",
+    "family_values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data[\"family\"] == value].groupby([\"date\"]).mean()\n",
```

Comment:
Optional: square brackets can be ommited in case of a single-column groupby:
```suggestion
    "  current_value_data = train_data[train_data[\"family\"] == value].groupby(\"date\").mean()\n",
```